cukinia

```diff
@@ -269,8 +269,7 @@ _cukinia_retry()
 	__retry=$((__retry-1))
 	sleep ${__wait_time:-0}
 	cukinia_runner $arguments
-
-	return $ret
+	ret=$?
```

Comment:
-1 : you should not rely on assumption for the ret variable existing, as it's likely to collide in some unsuspected ways. in the worst case, reserve a privately scoped variable (in the spirit of `__retry`) but ideally, considering _cukinia_retry sets up a recursive call chain this should be handled by recursive return value

Reply:
(unrelated to this change but I'd rather seek for a way to derecursive the retry feature, as we've seen it causes some versions of ash to crash as it's reaching its stack depth limit.)

Reply:
Acknowledge, directly returning `$?` instead of setting `ret` is better and more in the spirit of the recursion. I can make that modification if it's ok.

Regarding the recursive call I agree it's not ideal, I talked a bit about it to Kévin. Maybe this can be discussed in a separate issue ?

Reply:
yes, for the separate issue, it's only affecting specific versions of ash/dash (not sure which)